ql/src/java/org/apache/hadoop/hive/ql/ddl/table/info/show/tables/ShowTablesOperation.java

```diff
@@ -78,20 +75,22 @@ private void showTables() throws HiveException {
   }
 
   private void showTablesExtended() throws HiveException {
-    List<Table> tableObjects = new ArrayList<>();
-    tableObjects.addAll(context.getDb().getTableObjects(desc.getDbName(), null, desc.getTypeFilter()));
-    if (desc.getPattern() != null) {
-      Pattern pattern = Pattern.compile(UDFLike.likePatternToRegExp(desc.getPattern()), Pattern.CASE_INSENSITIVE);
-      tableObjects = tableObjects.stream()
-          .filter(object -> pattern.matcher(object.getTableName()).matches())
-          .collect(Collectors.toList());
+    List<Pair<String, String>> tableTypePairs = new ArrayList<>();
+    String pattern = MetaStoreUtils.convertSqlPatternToRegExp(desc.getPattern());
+    TableType typeFilter = desc.getTypeFilter();
+    TableType[] tableTypes = typeFilter == null ? TableType.values() : new TableType[]{typeFilter};
```

Comment:
why is this magic with typeFilter?

Reply:
It means to request all types table names if not input type filter in statement.